sdk/clientcore/core/src/main/java/io/clientcore/core/credentials/AccessToken.java

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package io.clientcore.core.credentials;
+
+import java.time.Duration;
+import java.time.OffsetDateTime;
+
+/**
+ * <p>
+ * Represents an immutable access token with a token string and an expiration time.
+ * </p>
+ *
+ * <p>
+ * Access Tokens are obtained through the authentication process, where the user or application presents valid
+ * credentials (such as a client ID, client secret, username/password, or certificate) to the authentication source.
+ * The authentication source then verifies the credentials and issues an Access Token, which is a time-limited token
+ * that grants access to the requested resource.
+ * </p>
+ *
+ * <p>
+ * Once an Access Token is obtained, it can be included in the Authorization header of HTTP requests to
+ * authenticate and authorize requests.
+ * </p>
+ *
+ * @see io.clientcore.core.credentials
+ * @see io.clientcore.core.credentials.TokenCredential
+ */
+public class AccessToken {
+    private final String token;
+    private final OffsetDateTime expiresAt;
+    private final OffsetDateTime refreshAt;
+    private final String tokenType;
+
+    /**
+     * Creates an access token instance.
+     *
+     * @param token the token string.
+     * @param expiresAt the expiration time.
+     */
+    public AccessToken(String token, OffsetDateTime expiresAt) {
```

Comment:
For all non-maximal overload constructors, it'd be really nice for the documentation to state that these will be `Bearer` tokens.